packages/analytics-core/src/plugins/destination.ts

```diff
@@ -297,21 +298,33 @@ export class Destination implements DestinationPlugin {
   }
 
   fulfillRequest(list: Context[], code: number, message: string) {
-    this.saveEvents();
     list.forEach((context) => context.callback(buildResult(context.event, code, message)));
+    void this.updateEventStorage(list);
   }
 
   /**
-   * Saves events to storage
-   * This is called on
-   * 1) new events are added to queue; or
-   * 2) response comes back for a request
+   * update the event storage
    */
-  saveEvents() {
+  async updateEventStorage(filterList: Context[], newContext?: Context[]) {
```

Comment:
imo the names are confusing as it doesn't tell what they are used for. can we rename `filterList` to `eventsToRemove and `newContext` to `eventsToAdd`? if I'm understanding it correctly